realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -572,37 +562,47 @@ public void emit(JavaWriter writer) throws IOException {
                 // LinkView currently does not support default value feature. Just fallback to normal code.
             }
         });
-        writer.emitStatement("proxyState.getRealm$realm().checkIfValid()");
-                if (Utils.isRealmModelList(field)) {
-                    writer.emitStatement("OsList osList = proxyState.getRow$realm().getModelList(%s)",
-                            fieldIndexVariableReference(field));
-                } else {
-                    writer.emitStatement("OsList osList = proxyState.getRow$realm().getValueList(%1$s, RealmFieldType.%2$s)",
-                            fieldIndexVariableReference(field), Utils.getValueListFieldType(field).name());
-                }
-                writer.emitStatement("osList.removeAll()")
-                .beginControlFlow("if (value == null)")
-                .emitStatement("return")
-                .endControlFlow();
 
+        writer.emitStatement("proxyState.getRealm$realm().checkIfValid()");
+        if (Utils.isRealmModelList(field)) {
+            writer.emitStatement("OsList osList = proxyState.getRow$realm().getModelList(%s)",
+                    fieldIndexVariableReference(field));
+        } else {
+            writer.emitStatement("OsList osList = proxyState.getRow$realm().getValueList(%1$s, RealmFieldType.%2$s)",
+                    fieldIndexVariableReference(field), Utils.getValueListFieldType(field).name());
+        }
         if (forRealmModel) {
-            writer.beginControlFlow("for (RealmModel linkedObject : value)")
-                    .beginControlFlow("if (!(RealmObject.isManaged(linkedObject) && RealmObject.isValid(linkedObject)))")
-                    .emitStatement("throw new IllegalArgumentException(\"Each element of 'value' must be a valid managed object.\")")
+            // Model lists.
+            writer
+                .beginControlFlow("if (value != null && value.size() == osList.size())")
```

Comment:
we can add the same comment as in https://github.com/realm/realm-java/pull/5396/files#diff-36f5b86acab52385e590ad92a6d7a80cR1294